syntax/printer.go

```diff
@@ -39,29 +49,39 @@ func SpaceRedirects(p *Printer) { p.spaceRedirects = true }
 // Note that this feature is best-effort and will only keep the
 // alignment stable, so it may need some human help the first time it is
 // run.
-func KeepPadding(p *Printer) {
-	p.keepPadding = true
-	p.cols.Writer = p.bufWriter.(*bufio.Writer)
-	p.bufWriter = &p.cols
+func KeepPadding() PrinterOption {
```

Comment:
Not convinced that it's worth swapping these to return a function. I assume this is for consistency and to make the godoc prettier?

Did you check if godoc groups all PrinterOptions, even if some of them are like `func KeepPadding(p *Printer) {`? If it doesn't, then I'd argue that's something to improve there. I don't see a problem per se in omitting parameters and a function return value when they're not needed.

This would also mean churn for the existing `syntax` and `interp` users, which further makes me think this is probably not something we want to do.

Reply:
>I assume this is for consistency and to make the godoc prettier?

Yes

>If it doesn't, then I'd argue that's something to improve there.

1. As I mentioned in issue, there is nothing godoc can do to group it without explicit types.
2. As I mentioned in commit message, these funcs should take bool argument
```
func BinaryNextLine(v bool) PrinterOption
func Indent(spaces uint) PrinterOption
func KeepPadding(v bool) PrinterOption
func Minify(v bool) PrinterOption
func SpaceRedirects(v bool) PrinterOption
func SwitchCaseIndent(v bool) PrinterOption
```
This way they become togglable. While I don't know if it's any useful for Printer, it definitely can be useful for Runner (one can imagine altering runner config with pseudocomands, like :set opt=foo when it's running interactively).

>This would also mean churn for the existing syntax and interp users, which further makes me think this is probably not something we want to do.

Since v3 is not stable yet why not take chance to improve docs and ease onboarding for all new users? Churn is unfortunate, yes, but this specific case can be quickly fixed with `sed s/syntax.BinaryNextLine/syntax.BinaryNextLine()/g`

One more example why I don't like top-level functions used as values:
take a look this popular package https://godoc.org/github.com/pkg/profile
I'm pretty sure it's impossible to quickly understand how to use it without taking a look at examples and I think this is a big enough problem. Explicit typing creates implicit documentation which needs zero maintenance. This is always a good thing.

Reply:
> This way they become togglable.

That's a good point, actually. I didn't think about disabling the options initially, because most of the time they're applied on a new object, where they default to false.

Please add the bool arguments and LGTM. Don't call them `v bool` though; perhaps `enabled bool`.

Reply:
Done